aea/manager.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the implementation of AEA agents manager."""
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+from aea.aea_builder import PathLike
+from aea.configurations.base import PublicId
+
+
+class AbstractManager(ABC):
+    """Abstract agents manager."""
+
+    def __init__(self, agents_dir: PathLike) -> None:
+        """
+        Initialize manager.
+
+        :param agents_dir: directory to store base agents.
+        """
+        self.agents_dir = agents_dir
+
+    @abstractmethod
+    def start_manager(self) -> None:
+        """Start manager."""
+
+    @abstractmethod
+    def stop_manager(self, cleanup: bool = False) -> None:
+        """
+        Stop manager.
+
+        Stops all running agents and stop agent.
+
+        :param cleanup: remove agents_dir and purge in memory registry.
+
+        :return: None
+        """
+
+    @abstractmethod
+    def fetch_agent(self, public_id: PublicId) -> None:
+        """Fetch agent and all dependencies to agents dir."""
+
+    @abstractmethod
+    def create_agent(
+        self, base_agent_id: PublicId, new_agent_id: PublicId, config_overrides: Dict
+    ) -> None:
```

Comment:
I don't think we need this like that. Basically, if you look at `tac-skills.md` you can see that there we create two identical agents `tac_participant_one` and `tac_participant_two`. However, here we don't need to create them in the file system twice (we only pull the original one `tac_participant`). Instead we just maintain them in memory using the `tac_participant_one` and `tac_participant_two` aliases to refer to them. In memory we have a dictionary which maps aliases (tac_participant_one/tac_participant_two etc) to AEAs.

Reply:
So it would become `add_agent(self, agent_id: PublicId, alias: str, config_overrides: Dict)`.